grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy

```diff
@@ -150,7 +151,8 @@ class Requisition implements Comparable<Requisition>, Serializable {
         statusSortOrder formula: RequisitionStatus.getStatusSortOrderFormula()
         monthRequested formula: "date_format(date_requested, '%M %Y')"
         comments joinTable: [name: "requisition_comment", key: "requisition_id"]
-        events joinTable: [name: "requisition_event", key: "requisition_id"]
+        // Fetch join in the events should be removed in Grails 3 and replaced with proper join query using data services
+        events joinTable: [name: "requisition_event", key: "requisition_id"], fetch: 'join'
```

Comment:
I was getting a `LazyInitializationException` when accessing the events in the `RequisitionStatusTransitionEvent` - it is because the event is done in a different session (after the session where we fetch the Requisition ends).

Current solution actually adds an eager fetching of the events, but using the `left join` clause, not calling a second query.
Normally I would not do it and would not recommend doing an eager fetch "globally", but I decided that it was not worth to spend time to create a custom query for that, because in Grails 1 we don't have **data services.** 
This is why I left a comment to refactor this ASAP in Grails 3, to use a proper joining query using **data services**

Reply:
@drodzewicz this would mean we’d have to write a pure SQL in order to fetch that. 
I just thought it’s waste of time for now, since we’ll migrate in a moment and shouldn’t experience any performance issues during that time

Reply:
I would probably try a withNewSession in the event handler. 
```
Requisition.withNewSession { 
    requisition = Requisition.read() // or Requisition.get() if you need to set something
}
```
Or if that doesn't work, just do a Requisition.read(id) or get(id) in the event handler. 

Lastly, you could potentially use refresh the event handler but that one always scares me.
```
requisition.refresh()
```
Not sure if you need the withSession or withNewSession wrapper so I'll leave that as an exercise for the reader.

Reply:
of course `get`/`read` will work, but I wanted to avoid additional query to refetch it, but if you are fine with it, I can just fetch it again in the event service.